fitbenchmarking/results_processing/plots.py

```diff
@@ -79,145 +74,59 @@ def write_html_with_link_plotlyjs(fig, figures_dir, htmlfile, options):
-            fig.add_trace(
-                go.Scatter(
-                    x=self.result.spinw_plot_info["ebin_cens"],
-                    y=df["y"][df["minimizer"] == "Starting Guess"][
-                        (data_len * i) : (data_len * (i + 1))
-                    ],
-                    name="Starting Guess",
-                    line=self._starting_guess_plot_line,
-                    showlegend=i == 0,
-                ),
-                row=1,
-                col=i + 1,
-            )
-            # plot data in both cases
-            fig.add_trace(
-                go.Scatter(
-                    x=self.result.spinw_plot_info["ebin_cens"],
-                    y=df["y"][df["minimizer"] == "Data"][
-                        (data_len * i) : (data_len * (i + 1))
-                    ],
-                    mode="markers",
-                    name="Data",
-                    marker=self._data_marker,
-                    showlegend=i == 0,
-                ),
-                row=1,
-                col=i + 1,
-            )
-
-        fig.update_layout(
-            legend={"yanchor": "auto", "y": 0.5, "xanchor": "right", "x": -0.1}
-        )
-        return fig
-
-    def plot_initial_guess(self, df):
+    def plot_initial_guess(self, df_fit) -> str:
         """
         Plots the initial guess along with the data and stores in a file
 
-        :param df: A dataframe holding the data
-        :type df: Pandas dataframe
+        :param df_fit: A dataframe holding the data
+        :type df_fit: Pandas dataframe
 
         :return: path to the saved file
         :rtype: str
         """
+        title = self.result.name
+        n_plots = 1
+        subplot_titles = None
+        ax_titles = self._default_ax_titles
 
-        if self.result.spinw_plot_info is not None:
-            fig = self.plotly_spinw(df)
-        else:
-            # Plotly implementation below
-            fig = px.line(
-                df[df["minimizer"] == "Starting Guess"],
-                x="x",
-                y="y",
-                color="minimizer",
-                title=self.result.name,
-                markers=True,
-            )
-            self._error_dict["array"] = df["e"][df["minimizer"] == "Data"]
-            # add the raw data as a scatter plot
-            fig.add_trace(
-                go.Scatter(
-                    x=df["x"][df["minimizer"] == "Data"],
-                    y=df["y"][df["minimizer"] == "Data"],
-                    error_y=self._error_dict,
-                    mode="markers",
-                    name="Data",
-                    marker=self._data_marker,
-                )
-            )
-            fig.update_layout(legend=self._legend_options)
+        if self.result.plot_info is not None:
+            n_plots = self.result.plot_info["n_plots"]
+            subplot_titles = self.result.plot_info["subplot_titles"]
+            ax_titles = self.result.plot_info["ax_titles"]
```

Comment:
Is it worth have a separate function for setting n_plots and titles, as these lines of code are repeated a few times in different functions?